flink/src/main/java/io/delta/flink/sink/internal/committer/DeltaGlobalCommitter.java

```diff
@@ -506,4 +607,17 @@ public void endOfInput() {
     @Override
     public void close() {
     }
+
+    private static class CheckpointData {
+
+        private final AddFile addFile;
+
+        private final DeltaCommittable committable;
```

Comment:
what do you use this for besides the checkpointId?

Reply:
To get DeltaPendingFile to get Partition Spec and Record count and File size (the last one probably could be taken from AddFile).

Record count and File size are used for Sink metrics, Partition Spec is used for this [check](https://github.com/kristoffSC/connectors/blob/a913e5d8d4790060f009e6ab8fd6f0ae81a4ee39/flink/src/main/java/io/delta/flink/sink/internal/committer/DeltaGlobalCommitter.java#:~:text=if%20(!isPartitionColumnsMetadataRetained)).

I'm not fully happy with current implementation, probably I will refactor it. I for "PoC" I wanted to limit the scope of changes of existing "commit transaction" logic without knowing if our design is correct.